src/main/java11/org/kohsuke/github/extras/HttpClientGitHubConnector.java

```diff
@@ -69,13 +87,50 @@ public GitHubConnectorResponse send(GitHubConnectorRequest connectorRequest) thr
         HttpRequest request = builder.build();
 
         try {
-            HttpResponse<InputStream> httpResponse = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+            HttpResponse<InputStream> httpResponse = send(request, 0);
             return new HttpClientGitHubConnectorResponse(connectorRequest, httpResponse);
         } catch (InterruptedException e) {
             throw (InterruptedIOException) new InterruptedIOException(e.getMessage()).initCause(e);
         }
     }
 
+    private HttpResponse<InputStream> send(HttpRequest request, int redirects)
```

Comment:
Okay, I see what you're trying to do here.  Looks good!    

But I'd like moved to `GitHubClient.detectRedirect`.   

From what you described, authorization stripping is the behavior GitHub expects. That means, regardless of what `GitHubConnector` implementation is used, when the connector returns a redirect response this project must strip the Authorization header as appropriate and then follow the redirect.  So, this functionality should go in `GitHubClient` instead of `HttpClientGitHubConnector`. 

On the plus side, `GitHubClient` already has functionality for retries. 

https://github.com/hub4j/github-api/blob/51b3a06fe364a9667d8c4e0b64abc780cff8d371/src/main/java/org/kohsuke/github/GitHubClient.java#L461-L471

A retry exception can be thrown during error detection and the updated request contained in the exception will be used for the next attempt. We use this for token refresh, now we can use it for redirects.   And we already detect redirects.  

https://github.com/hub4j/github-api/blob/51b3a06fe364a9667d8c4e0b64abc780cff8d371/src/main/java/org/kohsuke/github/GitHubClient.java#L508-L535

Use `detectExpiredToken` as the template and replace  `detectRedirect` with the logic you've implemented here.

Reply:
I dunno both if it's worth it and if it's safe enough.
As for the former, as I said, I did my homework and all the other clients are having this behavior of stripping the `Authorization` header. So sure, we could add another client that wouldn't have this behavior at some point in the future but I also think it's probably better to let the client handle the redirects if it can do it properly.

I also wanted to be very safe and carefully implement the rule of the Java HttpClient to make sure I wouldn't break anything.

Anyway, maybe you're right, I'm just not sure it's a decision that is so obvious.